packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -167,6 +203,101 @@ export class CanvasManager implements CanvasManagerInterface {
       unblockSelector,
       true,
     );
+    const rafId = this.snapshotCanvas(
+      false,
+      fps,
+      options,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+      undefined,
+      win,
+    );
+
+    this.resetObservers = () => {
+      canvasContextReset();
+      cancelAnimationFrame(rafId);
+    };
+  }
+
+  private initCanvasMutationObserver(
+    win: IWindow,
+    blockClass: blockClass,
+    blockSelector: string | null,
+    unblockSelector: string | null,
+  ): void {
+    this.startRAFTimestamping();
+    this.startPendingCanvasMutationFlusher();
+
+    const canvasContextReset = initCanvasContextObserver(
+      win,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+      false,
+    );
+    const canvas2DReset = initCanvas2DMutationObserver(
+      this.processMutation.bind(this),
+      win,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+    );
+
+    const canvasWebGL1and2Reset = initCanvasWebGLMutationObserver(
+      this.processMutation.bind(this),
+      win,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+      this.mirror,
+    );
+
+    this.resetObservers = () => {
+      canvasContextReset();
+      canvas2DReset();
+      canvasWebGL1and2Reset();
+    };
+  }
+
+  private manualSnapshot(
+    fps: number,
+    options: {
+      dataURLOptions: DataURLOptions;
+    },
+    blockClass: blockClass,
+    blockSelector: string | null,
+    unblockSelector: string | null,
+    canvasElement?: HTMLCanvasElement,
+  ) {
+    const rafId = this.snapshotCanvas(
+      true,
+      fps,
+      options,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+      canvasElement,
+    );
+
+    this.resetObservers = () => {
+      cancelAnimationFrame(rafId);
+    };
+  }
+
+  private snapshotCanvas(
```

Comment:
I would suggest making this more into a verb (also Canvas is a bit redundant):

```suggestion
  private takeSnapshot(
```